### PR TITLE
Performance optimization for SvgUnitConverter / SvgUnitCollectionConverter

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### Reference Issue

Split from #786

#### What does this implement/fix? Explain your changes.

Performance optimization for SvgUnitConverter / SvgUnitCollectionConverter

##### SvgUnitConverter

```bash
dotnet run -c Release -f netcoreapp3.1 -- -f '*SvgUnitConverter_*'
```

* master

|                       Method |     Mean |     Error |    StdDev |      Op/s | Rank |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------------- |---------:|----------:|----------:|----------:|-----:|-------:|------:|------:|----------:|
| SvgUnitConverter_ConvertFrom | 3.861 us | 0.0225 us | 0.0188 us | 259,008.8 |    1 | 0.5951 |     - |     - |   2.44 KB |

* PR

|                       Method |     Mean |     Error |    StdDev |      Op/s | Rank |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------------- |---------:|----------:|----------:|----------:|-----:|-------:|------:|------:|----------:|
| SvgUnitConverter_ConvertFrom | 3.059 us | 0.0507 us | 0.0450 us | 326,921.5 |    1 | 0.2747 |     - |     - |   1.13 KB |

After removing TypeConverter from picture  (which does allocate something when you call ConvertFrom) and using plain Parse we have no allocation in Parse method for SvgUnitConverter:

|                       Method |     Mean |     Error |    StdDev |      Op/s | Rank | Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------------- |---------:|----------:|----------:|----------:|-----:|------:|------:|------:|----------:|
| SvgUnitConverter_ConvertFrom | 2.846 us | 0.0057 us | 0.0048 us | 351,332.3 |    1 |     - |     - |     - |         - |

##### SvgUnitCollectionConverter and SvgStrokeDashArrayConverter

```bash
dotnet run -c Release -f netcoreapp3.1 -- -f '*SvgUnitCollectionConverter_*'
```

* master

|                           Method |     Mean |    Error |   StdDev |     Op/s | Rank |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------------------- |---------:|---------:|---------:|---------:|-----:|-------:|------:|------:|----------:|
| SvgUnitCollectionConverter_Parse | 23.94 us | 0.351 us | 0.311 us | 41,774.2 |    1 | 1.1597 |     - |     - |  21.29 KB |

* PR

|                           Method |     Mean |    Error |   StdDev |     Op/s | Rank |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------------------- |---------:|---------:|---------:|---------:|-----:|-------:|------:|------:|----------:|
| SvgUnitCollectionConverter_Parse | 18.80 us | 0.176 us | 0.138 us | 53,193.2 |    1 | 0.2747 |     - |     - |   5.44 KB |

### CODE REVIEW

**Comment by wieslawsoltes** [comment]
@mrbean-bremen This is ready for review and merge

**Review by mrbean-bremen — COMMENTED** [review]
Looks ok to me, but I would like to have someone more familiar with this kind of code have a look - probably @inforithmics.

**Comment by wieslawsoltes** [comment]
I will add some unit tests.

**Comment by wieslawsoltes** [comment]
@mrbean-bremen Added unit tests in https://github.com/svg-net/SVG/pull/811

File: Source/DataTypes/SvgUnitConverter.cs [hunk 1]
```diff
@@ -1,120 +1,193 @@
...
-            }
-
             // http://www.w3.org/TR/CSS21/syndata.html#values
             // http://www.w3.org/TR/SVG11/coords.html#Units
 
-            string unit = (string)value;
             int identifierIndex = -1;
 
-            if (unit == "none")
+            if (unit.SequenceEqual("none".AsSpan()))
```
Comment (inforithmics) [inline Source/DataTypes/SvgUnitConverter.cs:17]: wouldn't it be better to have a readonly variable with the value of

`private static readonly ReadOnlySpan<char> none= "none".AsSpan();`

and than compare 

`unit.SquenceEqual(none)`

File: Source/DataTypes/SvgUnitConverter.cs [hunk 2]
```diff
@@ -1,120 +1,193 @@
...
-            else if (unit == "large")
-                unit = "1.2em";
-            else if (unit == "x-large")
-                unit = "1.4em";
-            else if (unit == "xx-large")
-                unit = "1.7em";
-
-            for (int i = 0; i < unit.Length; i++)
+
+            if (unit.SequenceEqual("medium".AsSpan()))
```
Comment (inforithmics) [inline Source/DataTypes/SvgUnitConverter.cs:25]: `"medium".AsSpan()`  as constant

File: Source/DataTypes/SvgUnitConverter.cs (hunk 1 above)
Comment (wieslawsoltes) [reply Source/DataTypes/SvgUnitConverter.cs:17]: Not possible, you can only define ReadOnlySpan as field in `ref struct` not in class.

Compiler will give you error:
```
  SvgUnitConverter.cs(10, 33): [CS8345] Pole lub automatycznie implementowana właściwość nie może być typu „ReadOnlySpan<char>”, chyba że jest to składowa struktury ref.
```

**Comment by inforithmics** [comment]
It looks good for me. Nice work.

**Comment by wieslawsoltes** [comment]
@mrbean-bremen Marge master with unit tests and all pass.

**Pull request merged**